client/dashboard/data/error.ts

```diff
@@ -0,0 +1,13 @@
+export class DashboardDataError extends Error {
+	constructor(
+		public code: string,
+		cause?: unknown
+	) {
+		const message = cause instanceof Error ? cause.message : `Error: ${ cause }`;
+		super( message, { cause } );
+		this.name = 'DashboardDataError';
+
+		// Fix prototype chain (important for instanceof to work reliably)
+		Object.setPrototypeOf( this, new.target.prototype );
```

Comment:
AI suggested this. Apparently in some transpiling situations can break the prototype chain.